ntoskrnl/cc/view.c

```diff
@@ -647,20 +648,31 @@ CcRosLookupVacb (
                            VACB_MAPPING_GRANULARITY,
                            FileOffset))
         {
-            CcRosVacbIncRefCount(current);
-            KeReleaseSpinLockFromDpcLevel(&SharedCacheMap->CacheMapLock);
-            KeReleaseQueuedSpinLock(LockQueueMasterLock, oldIrql);
-            return current;
+            Ret = current;
+            break;
         }
+
         if (current->FileOffset.QuadPart > FileOffset)
             break;
+
         current_entry = current_entry->Flink;
     }
 
     KeReleaseSpinLockFromDpcLevel(&SharedCacheMap->CacheMapLock);
+
+    if (Ret)
```

Comment:
NOTE: Original code "equivalent" would have this `if (Ret) {...}` block just before the `KeReleaseSpinLockFromDpcLevel(&SharedCacheMap->CacheMapLock);`

Reply:
The cache map lock is not needed for modifying the LRU list (see `CcRosCreateVacb`).